posts/tree-cell-typing/index.qmd

```diff
@@ -0,0 +1,241 @@
+- @sec-additional-recommendations Additional recommendations
+- @sec-conclusions Conclusions
+
+# InSituTree Overview: How does it work? {#sec-package-overview}
+
+This package is an extension for the `InSituType` package 
+and takes a nested approach to perform hierarchical supervised cell typing. 
+Rather than immediately 
+trying to assign cells to all subtypes available, it first starts by classifying
+cells into broad categories and then assigns cells to finer subtypes.
+
+There are two core concepts that underlie this tree approach and differentiate 
+it from the standard supervised `InSituType`:
+
+1. **Hierarchical annotation** Here, we break up the cell typing into a series 
+of annotation decisions, starting at coarse cell type categories and proceeding 
+down to fine categories
+    + Example: Immune -> Lymphoid -> T-cell -> T4 -> Treg
+2. **Automatic feature selection** The genes used for each decision point are 
+optimized for maximum contrast given the cell types under consideration.
+
+Aligned with these two core concepts, there are two additional types of input 
+that need to be provided to the function call beyond what is required for 
+supervised `InSituType`:
+
+* `cth`:  **c**ell **t**ype **h**ierarchy, a nested list object that describes 
+the relationship between cell types
+* `quantile_absolute_expression_difference_param` and `quantile_percent_expression_difference_param`: feature selection parameters 
+based on reference profile expression of each gene at absolute level and 
+percentile across the entire dataset.
+
+# `InSituTree` Example: Running the package {#sec-example-implementation}
+
+You can find the `InSituTree` package [here](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/_code/InSituTree){target="_blank"}. See the corresponding [vignette](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/_code/InSituTree/vignettes/InSituTree_basicVignette.html){target="_blank"} inside the package for more details. 
+
+For this example we work with the immune-focused cell type profiles 
+as well as the mini non-small cell lung cancer dataset included
+in the `InSituType` package. 
+
+```{r setup, message=FALSE, eval = FALSE}
+# Load libraries
+library(InSituType)
+library(InSituTree)
+
+# Read in io profiles
+data(ioprofiles)
+
+head(ioprofiles)
+```
+
+```{r, eval = TRUE, echo = FALSE}
+profile_head <- readRDS("./assets/io_head.rds")
+head(profile_head)
+```
+
+We start by setting up a nested list of cell types,
+visualized with the helper function `printTree()`.
+The nested list should have names for major cell types (for example,
+"B-lymphoid" or "structural" below), and should include all cell types in the
+reference profile as endpoints (for example, "B-cell" or "endothelial" below).
+The endpoint cell types do not need to be set up as lists (for example, see "NK"
+below which is not set as "NK" = "NK"), though setting them as lists will not
```

Comment:
how about this: 
`The endpoint cell types could be set as vector instead of a list (see "NK" below which is set as unnamed character vector of 1 length), though setting them as lists will not cause errors. `

Reply:
Modified, good suggestion.